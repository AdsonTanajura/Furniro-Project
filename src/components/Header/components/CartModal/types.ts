export type CartModalProps = {
  setIsCartOpen: (v: boolean) => void;
};
