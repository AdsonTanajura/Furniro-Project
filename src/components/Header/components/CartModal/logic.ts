import { useNavigate } from 'react-router-dom';
import CartModalProps from './interfaces/CartModalProps';

export const useCartModalLogic = ({ setIsCartOpen }: CartModalProps) => {
  const navigate = useNavigate();

  const handleClickCloseButton = () => {
    setIsCartOpen(false);
  };
  const handleCheckoutButton = () => {
    navigate('home/checkout');
    setIsCartOpen(false);
  };

  const handleCartButton = () => {
    navigate('home/cart');
    setIsCartOpen(false);
  };

  return {
    handleCartButton,
    handleCheckoutButton,
    handleClickCloseButton,
  };
};
