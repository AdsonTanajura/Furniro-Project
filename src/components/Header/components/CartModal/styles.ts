import { styled } from 'styled-components';

export const Container = styled.div`
  position: absolute;
  background-color: #ffffff;
  width: 240px;
  z-index: 20;
  right: 0;
  top: 0;
  height: 100vh;

  @media (min-width: 768px) {
    width: auto;
  }
`;

export const ProductContainer = styled.div`
  display: flex;
  flex-direction: column;
  padding: 14px 20px;

  @media (min-width: 768px) {
    padding: 28px 40px 24px 28px;
  }
`;

export const TitleContainer = styled.div`
  display: flex;
  justify-content: center;
  flex-direction: row;
  gap: 8px;

  @media (min-width: 768px) {
    gap: 42px;
  }
`;

export const Title = styled.h2`
  font-size: 24px;
  font-weight: 600;
`;

export const TitleBorderContainer = styled.div`
  flex-direction: row;
  padding-bottom: 26px;
  border-bottom: 2px solid #d9d9d9;

  @media (min-width: 768px) {
    min-width: 287px;
  }
`;

export const CloseButton = styled.button`
  background-color: transparent;
  border: none;
  padding: 2px 4px;
  color: #9f9f9f;
  font-size: 20px;

  &:hover {
    color: black;
  }
`;

export const ButtonContainer = styled.div`
  display: flex;
  flex-direction: row;
  border-top: 2px solid #d9d9d9;
  padding: 13px 13px;
  gap: 8px;

  @media (min-width: 768px) {
    padding: 26px 26px;
    gap: 14px;
  }
`;

export const Button = styled.button`
  background-color: transparent;
  border: 2px solid #000000;
  border-radius: 50px;
  padding: 6px 6px;
  transition: all 0.2s;
  font-size: 12px;

  @media (min-width: 768px) {
    padding: 6px 30px;
  }

  &:hover {
    background-color: #000000;
    color: #fff;
  }
`;
