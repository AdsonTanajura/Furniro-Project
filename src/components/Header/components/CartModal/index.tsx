import {
  Button,
  ButtonContainer,
  CloseButton,
  CartModalContainer,
  ProductContainer,
  Title,
  TitleBorderContainer,
  TitleContainer,
} from './styles';
import CardProductItem from '../CardProductItem';
import {
  ProductList,
  Subtotal,
  SubtotalContainer,
  SubtotalPrice,
} from '../CardProductItem/styles';
import { FaWindowClose } from 'react-icons/fa';
import { useAppSelector } from '../../../../hooks';
import formatCurrencyRp from '../../../../utils/formatCurrencyRp';
import CartModalProps from './interfaces/CartModalProps';
import { useCartModalLogic } from './logic';

const CartModal = ({ setIsCartOpen }: CartModalProps) => {
  const { handleCartButton, handleCheckoutButton, handleClickCloseButton } =
    useCartModalLogic({ setIsCartOpen });

  const cart = useAppSelector((state) => state.cart);
  return (
    <CartModalContainer>
      <ProductContainer>
        <TitleContainer>
          <TitleBorderContainer>
            <Title>Shopping Cart</Title>
          </TitleBorderContainer>
          <div>
            <CloseButton
              data-testid="close-button"
              onClick={handleClickCloseButton}
            >
              <FaWindowClose />
            </CloseButton>
          </div>
        </TitleContainer>
        <ProductList>
          {cart.items.map((item) => (
            <CardProductItem
              productImage={item.image}
              productPrice={item.price}
              productQuantity={item.quantity}
              productTitle={item.name}
              productId={item.id}
              key={item.id}
            />
          ))}
        </ProductList>
        <SubtotalContainer>
          <Subtotal>Subtotal</Subtotal>
          <SubtotalPrice>{formatCurrencyRp(cart.subtotal)}</SubtotalPrice>
        </SubtotalContainer>
      </ProductContainer>
      <ButtonContainer>
        <Button onClick={handleCartButton}>Cart</Button>
        <Button onClick={handleCheckoutButton}>Checkout</Button>
        <Button>Comparison</Button>
      </ButtonContainer>
    </CartModalContainer>
  );
};

export default CartModal;
