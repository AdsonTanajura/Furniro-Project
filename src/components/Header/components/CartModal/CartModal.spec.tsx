import { render } from '@testing-library/react';
import CartModal from '.';
import { MemoryRouter, useNavigate } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from '../../../../store';
import userEvent from '@testing-library/user-event';

const mockSetIsCartOpen = jest.fn();

jest.mock('react-router-dom', () => {
  const actual = jest.requireActual('react-router-dom');
  return {
    ...actual,
    useNavigate: jest.fn(),
  };
});

const mockNavigate = jest.fn();
(useNavigate as jest.Mock).mockReturnValue(mockNavigate);

describe('<CartModal />', () => {
  it('should take you to the cart page when you click on the cart button', async () => {
    const { container, getByRole } = render(
      <Provider store={store}>
        <MemoryRouter>
          <CartModal setIsCartOpen={mockSetIsCartOpen} />
        </MemoryRouter>
      </Provider>
    );

    const cartButton = getByRole('button', { name: /cart/i });

    expect(cartButton).toBeInTheDocument();
    await userEvent.click(cartButton);

    expect(mockNavigate).toHaveBeenCalledWith('home/cart');

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should take you to the Checkout page when you click on the Checkout button', async () => {
    const { container, getByRole } = render(
      <Provider store={store}>
        <MemoryRouter>
          <CartModal setIsCartOpen={mockSetIsCartOpen} />
        </MemoryRouter>
      </Provider>
    );

    const cartButton = getByRole('button', { name: /checkout/i });

    expect(cartButton).toBeInTheDocument();
    await userEvent.click(cartButton);

    expect(mockNavigate).toHaveBeenCalledWith('home/checkout');

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should return false in the mockSetIsCartOpen when clicking the close button', async () => {
    const { container, getByTestId } = render(
      <Provider store={store}>
        <MemoryRouter>
          <CartModal setIsCartOpen={mockSetIsCartOpen} />
        </MemoryRouter>
      </Provider>
    );

    const closeButton = getByTestId('close-button');

    expect(closeButton).toBeInTheDocument();
    expect(mockSetIsCartOpen).not.toHaveBeenCalled();
    await userEvent.click(closeButton);
    expect(mockSetIsCartOpen).toHaveBeenCalled();
    expect(mockSetIsCartOpen).toHaveBeenCalledWith(false);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the main elements', () => {
    const { container, getAllByRole } = render(
      <Provider store={store}>
        <MemoryRouter>
          <CartModal setIsCartOpen={mockSetIsCartOpen} />
        </MemoryRouter>
      </Provider>
    );
    const buttons = getAllByRole('button');
    const titles = getAllByRole('heading', { level: 2 });

    expect(titles.length).toBeGreaterThan(0);
    titles.forEach((title) => expect(title).toBeInTheDocument());
    titles.forEach((title) => expect(title).toBeVisible());

    expect(buttons.length).toBeGreaterThan(0);
    buttons.forEach((button) => expect(button).toBeInTheDocument());
    buttons.forEach((button) => expect(button).toBeVisible());

    expect(container.firstChild).toBeInTheDocument();
    expect(container.firstChild).toBeVisible();
    expect(container.firstChild).toMatchSnapshot();
  });
});
