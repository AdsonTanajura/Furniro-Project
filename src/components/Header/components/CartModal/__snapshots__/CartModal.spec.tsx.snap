// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartModal /> should render the main elements 1`] = `
.c0 {
  position: absolute;
  background-color: #ffffff;
  width: 240px;
  z-index: 20;
  right: 0;
  top: 0;
  height: 100vh;
}

.c1 {
  display: flex;
  flex-direction: column;
  padding: 14px 20px;
}

.c2 {
  display: flex;
  justify-content: center;
  flex-direction: row;
  gap: 8px;
}

.c4 {
  font-size: 24px;
  font-weight: 600;
}

.c3 {
  flex-direction: row;
  padding-bottom: 26px;
  border-bottom: 2px solid #d9d9d9;
}

.c5 {
  background-color: transparent;
  border: none;
  padding: 2px 4px;
  color: #9f9f9f;
  font-size: 20px;
}

.c5:hover {
  color: black;
}

.c10 {
  display: flex;
  flex-direction: row;
  border-top: 2px solid #d9d9d9;
  padding: 13px 13px;
  gap: 8px;
}

.c11 {
  background-color: transparent;
  border: 2px solid #000000;
  border-radius: 50px;
  padding: 6px 6px;
  transition: all 0.2s;
  font-size: 12px;
}

.c11:hover {
  background-color: #000000;
  color: #fff;
}

.c6 {
  display: flex;
  overflow-y: scroll;
  flex-direction: column;
  margin: 42px 0px;
  gap: 20px;
  height: calc(100vh - 300px);
}

.c7 {
  display: flex;
  flex-direction: row;
  gap: 25px;
}

.c8 {
  font-size: 16px;
  font-weight: 400;
}

.c9 {
  color: #b88e2f;
  font-size: 16px;
  font-weight: 600;
}

@media (min-width: 768px) {
  .c0 {
    width: auto;
  }
}

@media (min-width: 768px) {
  .c1 {
    padding: 28px 40px 24px 28px;
  }
}

@media (min-width: 768px) {
  .c2 {
    gap: 42px;
  }
}

@media (min-width: 768px) {
  .c3 {
    min-width: 287px;
  }
}

@media (min-width: 768px) {
  .c10 {
    padding: 26px 26px;
    gap: 14px;
  }
}

@media (min-width: 768px) {
  .c11 {
    padding: 6px 30px;
  }
}

@media (min-width: 768px) {
  .c7 {
    gap: 101px;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <h2
          class="c4"
        >
          Shopping Cart
        </h2>
      </div>
      <div>
        <button
          class="c5"
          data-testid="close-button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M464 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-83.6 290.5c4.8 4.8 4.8 12.6 0 17.4l-40.5 40.5c-4.8 4.8-12.6 4.8-17.4 0L256 313.3l-66.5 67.1c-4.8 4.8-12.6 4.8-17.4 0l-40.5-40.5c-4.8-4.8-4.8-12.6 0-17.4l67.1-66.5-67.1-66.5c-4.8-4.8-4.8-12.6 0-17.4l40.5-40.5c4.8-4.8 12.6-4.8 17.4 0l66.5 67.1 66.5-67.1c4.8-4.8 12.6-4.8 17.4 0l40.5 40.5c4.8 4.8 4.8 12.6 0 17.4L313.3 256l67.1 66.5z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="c6"
    />
    <div
      class="c7"
    >
      <span
        class="c8"
      >
        Subtotal
      </span>
      <span
        class="c9"
      >
        Rp 0
      </span>
    </div>
  </div>
  <div
    class="c10"
  >
    <button
      class="c11"
    >
      Cart
    </button>
    <button
      class="c11"
    >
      Checkout
    </button>
    <button
      class="c11"
    >
      Comparison
    </button>
  </div>
</div>
`;

exports[`<CartModal /> should return false in the mockSetIsCartOpen when clicking the close button 1`] = `
.c0 {
  position: absolute;
  background-color: #ffffff;
  width: 240px;
  z-index: 20;
  right: 0;
  top: 0;
  height: 100vh;
}

.c1 {
  display: flex;
  flex-direction: column;
  padding: 14px 20px;
}

.c2 {
  display: flex;
  justify-content: center;
  flex-direction: row;
  gap: 8px;
}

.c4 {
  font-size: 24px;
  font-weight: 600;
}

.c3 {
  flex-direction: row;
  padding-bottom: 26px;
  border-bottom: 2px solid #d9d9d9;
}

.c5 {
  background-color: transparent;
  border: none;
  padding: 2px 4px;
  color: #9f9f9f;
  font-size: 20px;
}

.c5:hover {
  color: black;
}

.c10 {
  display: flex;
  flex-direction: row;
  border-top: 2px solid #d9d9d9;
  padding: 13px 13px;
  gap: 8px;
}

.c11 {
  background-color: transparent;
  border: 2px solid #000000;
  border-radius: 50px;
  padding: 6px 6px;
  transition: all 0.2s;
  font-size: 12px;
}

.c11:hover {
  background-color: #000000;
  color: #fff;
}

.c6 {
  display: flex;
  overflow-y: scroll;
  flex-direction: column;
  margin: 42px 0px;
  gap: 20px;
  height: calc(100vh - 300px);
}

.c7 {
  display: flex;
  flex-direction: row;
  gap: 25px;
}

.c8 {
  font-size: 16px;
  font-weight: 400;
}

.c9 {
  color: #b88e2f;
  font-size: 16px;
  font-weight: 600;
}

@media (min-width: 768px) {
  .c0 {
    width: auto;
  }
}

@media (min-width: 768px) {
  .c1 {
    padding: 28px 40px 24px 28px;
  }
}

@media (min-width: 768px) {
  .c2 {
    gap: 42px;
  }
}

@media (min-width: 768px) {
  .c3 {
    min-width: 287px;
  }
}

@media (min-width: 768px) {
  .c10 {
    padding: 26px 26px;
    gap: 14px;
  }
}

@media (min-width: 768px) {
  .c11 {
    padding: 6px 30px;
  }
}

@media (min-width: 768px) {
  .c7 {
    gap: 101px;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <h2
          class="c4"
        >
          Shopping Cart
        </h2>
      </div>
      <div>
        <button
          class="c5"
          data-testid="close-button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M464 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-83.6 290.5c4.8 4.8 4.8 12.6 0 17.4l-40.5 40.5c-4.8 4.8-12.6 4.8-17.4 0L256 313.3l-66.5 67.1c-4.8 4.8-12.6 4.8-17.4 0l-40.5-40.5c-4.8-4.8-4.8-12.6 0-17.4l67.1-66.5-67.1-66.5c-4.8-4.8-4.8-12.6 0-17.4l40.5-40.5c4.8-4.8 12.6-4.8 17.4 0l66.5 67.1 66.5-67.1c4.8-4.8 12.6-4.8 17.4 0l40.5 40.5c4.8 4.8 4.8 12.6 0 17.4L313.3 256l67.1 66.5z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="c6"
    />
    <div
      class="c7"
    >
      <span
        class="c8"
      >
        Subtotal
      </span>
      <span
        class="c9"
      >
        Rp 0
      </span>
    </div>
  </div>
  <div
    class="c10"
  >
    <button
      class="c11"
    >
      Cart
    </button>
    <button
      class="c11"
    >
      Checkout
    </button>
    <button
      class="c11"
    >
      Comparison
    </button>
  </div>
</div>
`;

exports[`<CartModal /> should take you to the Checkout page when you click on the Checkout button 1`] = `
.c0 {
  position: absolute;
  background-color: #ffffff;
  width: 240px;
  z-index: 20;
  right: 0;
  top: 0;
  height: 100vh;
}

.c1 {
  display: flex;
  flex-direction: column;
  padding: 14px 20px;
}

.c2 {
  display: flex;
  justify-content: center;
  flex-direction: row;
  gap: 8px;
}

.c4 {
  font-size: 24px;
  font-weight: 600;
}

.c3 {
  flex-direction: row;
  padding-bottom: 26px;
  border-bottom: 2px solid #d9d9d9;
}

.c5 {
  background-color: transparent;
  border: none;
  padding: 2px 4px;
  color: #9f9f9f;
  font-size: 20px;
}

.c5:hover {
  color: black;
}

.c10 {
  display: flex;
  flex-direction: row;
  border-top: 2px solid #d9d9d9;
  padding: 13px 13px;
  gap: 8px;
}

.c11 {
  background-color: transparent;
  border: 2px solid #000000;
  border-radius: 50px;
  padding: 6px 6px;
  transition: all 0.2s;
  font-size: 12px;
}

.c11:hover {
  background-color: #000000;
  color: #fff;
}

.c6 {
  display: flex;
  overflow-y: scroll;
  flex-direction: column;
  margin: 42px 0px;
  gap: 20px;
  height: calc(100vh - 300px);
}

.c7 {
  display: flex;
  flex-direction: row;
  gap: 25px;
}

.c8 {
  font-size: 16px;
  font-weight: 400;
}

.c9 {
  color: #b88e2f;
  font-size: 16px;
  font-weight: 600;
}

@media (min-width: 768px) {
  .c0 {
    width: auto;
  }
}

@media (min-width: 768px) {
  .c1 {
    padding: 28px 40px 24px 28px;
  }
}

@media (min-width: 768px) {
  .c2 {
    gap: 42px;
  }
}

@media (min-width: 768px) {
  .c3 {
    min-width: 287px;
  }
}

@media (min-width: 768px) {
  .c10 {
    padding: 26px 26px;
    gap: 14px;
  }
}

@media (min-width: 768px) {
  .c11 {
    padding: 6px 30px;
  }
}

@media (min-width: 768px) {
  .c7 {
    gap: 101px;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <h2
          class="c4"
        >
          Shopping Cart
        </h2>
      </div>
      <div>
        <button
          class="c5"
          data-testid="close-button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M464 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-83.6 290.5c4.8 4.8 4.8 12.6 0 17.4l-40.5 40.5c-4.8 4.8-12.6 4.8-17.4 0L256 313.3l-66.5 67.1c-4.8 4.8-12.6 4.8-17.4 0l-40.5-40.5c-4.8-4.8-4.8-12.6 0-17.4l67.1-66.5-67.1-66.5c-4.8-4.8-4.8-12.6 0-17.4l40.5-40.5c4.8-4.8 12.6-4.8 17.4 0l66.5 67.1 66.5-67.1c4.8-4.8 12.6-4.8 17.4 0l40.5 40.5c4.8 4.8 4.8 12.6 0 17.4L313.3 256l67.1 66.5z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="c6"
    />
    <div
      class="c7"
    >
      <span
        class="c8"
      >
        Subtotal
      </span>
      <span
        class="c9"
      >
        Rp 0
      </span>
    </div>
  </div>
  <div
    class="c10"
  >
    <button
      class="c11"
    >
      Cart
    </button>
    <button
      class="c11"
    >
      Checkout
    </button>
    <button
      class="c11"
    >
      Comparison
    </button>
  </div>
</div>
`;

exports[`<CartModal /> should take you to the cart page when you click on the cart button 1`] = `
.c0 {
  position: absolute;
  background-color: #ffffff;
  width: 240px;
  z-index: 20;
  right: 0;
  top: 0;
  height: 100vh;
}

.c1 {
  display: flex;
  flex-direction: column;
  padding: 14px 20px;
}

.c2 {
  display: flex;
  justify-content: center;
  flex-direction: row;
  gap: 8px;
}

.c4 {
  font-size: 24px;
  font-weight: 600;
}

.c3 {
  flex-direction: row;
  padding-bottom: 26px;
  border-bottom: 2px solid #d9d9d9;
}

.c5 {
  background-color: transparent;
  border: none;
  padding: 2px 4px;
  color: #9f9f9f;
  font-size: 20px;
}

.c5:hover {
  color: black;
}

.c10 {
  display: flex;
  flex-direction: row;
  border-top: 2px solid #d9d9d9;
  padding: 13px 13px;
  gap: 8px;
}

.c11 {
  background-color: transparent;
  border: 2px solid #000000;
  border-radius: 50px;
  padding: 6px 6px;
  transition: all 0.2s;
  font-size: 12px;
}

.c11:hover {
  background-color: #000000;
  color: #fff;
}

.c6 {
  display: flex;
  overflow-y: scroll;
  flex-direction: column;
  margin: 42px 0px;
  gap: 20px;
  height: calc(100vh - 300px);
}

.c7 {
  display: flex;
  flex-direction: row;
  gap: 25px;
}

.c8 {
  font-size: 16px;
  font-weight: 400;
}

.c9 {
  color: #b88e2f;
  font-size: 16px;
  font-weight: 600;
}

@media (min-width: 768px) {
  .c0 {
    width: auto;
  }
}

@media (min-width: 768px) {
  .c1 {
    padding: 28px 40px 24px 28px;
  }
}

@media (min-width: 768px) {
  .c2 {
    gap: 42px;
  }
}

@media (min-width: 768px) {
  .c3 {
    min-width: 287px;
  }
}

@media (min-width: 768px) {
  .c10 {
    padding: 26px 26px;
    gap: 14px;
  }
}

@media (min-width: 768px) {
  .c11 {
    padding: 6px 30px;
  }
}

@media (min-width: 768px) {
  .c7 {
    gap: 101px;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <h2
          class="c4"
        >
          Shopping Cart
        </h2>
      </div>
      <div>
        <button
          class="c5"
          data-testid="close-button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M464 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-83.6 290.5c4.8 4.8 4.8 12.6 0 17.4l-40.5 40.5c-4.8 4.8-12.6 4.8-17.4 0L256 313.3l-66.5 67.1c-4.8 4.8-12.6 4.8-17.4 0l-40.5-40.5c-4.8-4.8-4.8-12.6 0-17.4l67.1-66.5-67.1-66.5c-4.8-4.8-4.8-12.6 0-17.4l40.5-40.5c4.8-4.8 12.6-4.8 17.4 0l66.5 67.1 66.5-67.1c4.8-4.8 12.6-4.8 17.4 0l40.5 40.5c4.8 4.8 4.8 12.6 0 17.4L313.3 256l67.1 66.5z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="c6"
    />
    <div
      class="c7"
    >
      <span
        class="c8"
      >
        Subtotal
      </span>
      <span
        class="c9"
      >
        Rp 0
      </span>
    </div>
  </div>
  <div
    class="c10"
  >
    <button
      class="c11"
    >
      Cart
    </button>
    <button
      class="c11"
    >
      Checkout
    </button>
    <button
      class="c11"
    >
      Comparison
    </button>
  </div>
</div>
`;
